utilities/test_suite/HIP/runTests.py

```diff
@@ -291,9 +498,184 @@ def rpp_test_suite_parser_and_validator():
+                                    process = subprocess.Popen([ 'rocprof', '--basenames', 'on', '--timestamp', 'on', '--stats', '-o', f"{dstPath}/Tensor_PKD3/case_{case}/output_case{case}_bitDepth{bitDepth}_oft{outputFormatToggle}.csv", './Tensor_hip', srcPath1, srcPath2, dstPath, str(bitDepth), str(outputFormatToggle), str(case), '0', str(numRuns), str(testType), str(layout), '0', str(qaMode), str(decoderType), str(batchSize), str(userRoi)] + roiList, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+                                    while True:
+                                        output = process.stdout.readline()
+                                        if not output and process.poll() is not None:
+                                            break
+                                        print(output.strip())
+                                        log_file.write(output.decode('utf-8'))
+                            elif layout == 1:
+                                if not os.path.exists(f"{dstPath}/Tensor_PLN3/case_{case}"):
+                                    os.mkdir(f"{dstPath}/Tensor_PLN3/case_{case}")
+                                with open(f"{loggingFolder}/Tensor_hip_{log_file_layout}_raw_performance_log.txt", "a") as log_file:
+                                    print(f"\nrocprof --basenames on --timestamp on --stats -o {dstPath}/Tensor_PLN1/case_{case}/output_case{case}_bitDepth{bitDepth}_oft{outputFormatToggle}.csv ./Tensor_hip {srcPath1} {srcPath2} {dstPath} {bitDepth} {outputFormatToggle} {case} 0 0")
+                                    process = subprocess.Popen([ 'rocprof', '--basenames', 'on', '--timestamp', 'on', '--stats', '-o', f"{dstPath}/Tensor_PLN1/case_{case}/output_case{case}_bitDepth{bitDepth}_oft{outputFormatToggle}.csv", './Tensor_hip', srcPath1, srcPath2, dstPath, str(bitDepth), str(outputFormatToggle), str(case), '0', str(numRuns), str(testType), str(layout), '0', str(qaMode), str(decoderType), str(batchSize), str(userRoi)] + roiList, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+                                    while True:
+                                        output = process.stdout.readline()
+                                        if not output and process.poll() is not None:
+                                            break
+                                        print(output.strip())
+                                        log_file.write(output.decode('utf-8'))
+                            elif layout == 2:
+                                if not os.path.exists(f"{dstPath}/Tensor_PLN1/case_{case}"):
+                                    os.mkdir(f"{dstPath}/Tensor_PLN1/case_{case}")
+                                with open(f"{loggingFolder}/Tensor_hip_{log_file_layout}_raw_performance_log.txt", "a") as log_file:
```

Comment:
The code in python script has increased a lot
is there no scope for modularization?